File: template-method/typescript/html-data-analyzer.ts
import {DataAnalyzerBase} from "./data-analyzer-base";

// UML diyagramındaki AbstractClass (DataAnalyzerBase) sınıfından türer.
// UML diyagramındaki ConcreteClass yapısına denk gelmektedir.
export class HtmlDataAnalyzer extends DataAnalyzerBase {
  public exportResult(result: object): void {
    // Durumun HTML'e dönüştürülmesi işlemleri yer alacaktır.
    console.log("HTML exported.");
  }
}

File: template-method/go/html-data-analyzer.go
package main

import "fmt"

// UML diyagramındaki AbstractClass (DataAnalyzerBase) sınıfından türer.
// UML diyagramındaki ConcreteClass yapısına denk gelmektedir.
type htmlDataAnalyzer struct {
	baseAnalyzer *DataAnalyzerBase
}

func NewHTMLDataAnalyzer(b *DataAnalyzerBase) *htmlDataAnalyzer {
	return &htmlDataAnalyzer{
		baseAnalyzer: b,
	}
}

func (h htmlDataAnalyzer) exportResult(_ []byte) {
	// Durumun HTML'e dönüştürülmesi işlemleri yer alacaktır.e
	fmt.Println("HTML exported.")
}
